src/views/virtualmachines/tree/hooks/useTreeViewData.ts

```diff
@@ -85,57 +99,70 @@ export const useTreeViewData = (): UseTreeViewData => {
   const memoizedVMIMs = useMemo(
     () =>
       getLatestMigrationForEachVM(
-        isAdmin
+        // is the || really necessary here?
+        isAdmin || supportsMulticluster
           ? allVMIM
           : Object.values(allowedVMIMResources).flatMap((resource) => resource.data),
       ),
-    [allVMIM, allowedVMIMResources, isAdmin],
+    [allVMIM, allowedVMIMResources, isAdmin, supportsMulticluster],
   );
 
   vmimMapperSignal.value = memoizedVMIMs;
 
   const memoizedVMs = useMemo(
-    () => (isAdmin ? allVMs : Object.values(allowedResources).flatMap((resource) => resource.data)),
-    [allVMs, allowedResources, isAdmin],
+    () =>
+      // is the || really necessary here?
+      isAdmin || supportsMulticluster
+        ? allVMs
+        : Object.values(allowedResources).flatMap((resource) => resource.data),
+    [allVMs, allowedResources, isAdmin, supportsMulticluster],
   );
-
   vmsSignal.value = memoizedVMs;
 
-  const getResourceUrl = useGetResourceUrl();
-
   const loaded =
     projectNamesLoaded &&
     (isAdmin ? allVMsLoaded : Object.values(allowedResources).some((resource) => resource.loaded));
-
-  const treeData = useMemo(
-    () =>
-      loaded
-        ? createTreeViewData(
-            getResourceUrl,
-            projectNames,
-            memoizedVMs,
-            isAdmin,
-            location.pathname,
-            treeViewFoldersEnabled,
-          )
-        : [],
-    [
-      loaded,
-      getResourceUrl,
-      projectNames,
-      memoizedVMs,
-      isAdmin,
-      location.pathname,
-      treeViewFoldersEnabled,
-    ],
-  );
-
-  const isSwitchDisabled = useMemo(() => projectNames.every(isSystemNamespace), [projectNames]);
+  const loadedMulticluster = multiclusterVmLoaded && namespaceNamesLoaded;
+
+  const memoizedtreeData = useMemo(() => {
+    const aggrigateClusterTreeData = [];
+    const activeProjectList = supportsMulticluster ? namespaceNamesByCluster : projectNames;
+    const activeVMsList = supportsMulticluster ? multiclusterVms : memoizedVMs;
+    if ((loaded && !supportsMulticluster) || loadedMulticluster) {
+      aggrigateClusterTreeData.push(
+        ...createTreeViewData(
+          getResourceUrlOverride,
+          isAdmin,
+          location.pathname,
+          treeViewFoldersEnabled,
+          activeProjectList,
+          activeVMsList,
+          supportsMulticluster,
+        ),
+      );
+    }
+    return aggrigateClusterTreeData;
+  }, [
+    supportsMulticluster,
+    namespaceNamesByCluster,
+    projectNames,
+    multiclusterVms,
+    memoizedVMs,
+    loaded,
+    loadedMulticluster,
+    getResourceUrlOverride,
+    isAdmin,
+    location.pathname,
+    treeViewFoldersEnabled,
+  ]);
```

Comment:
Revisit this code... it should be able to resemble the original much more closely, after you change `allVMs` to include multicluster data, rather than adding a separate call for that.